lang/src/org/partiql/lang/planner/transforms/AstToLogicalVisitorTransform.kt

```diff
@@ -507,6 +587,56 @@ private class FromSourceToBexpr(
         }
 }
 
+/**
+ * Given a [PartiqlAst.Expr.Select], transforms all [PartiqlAst.Expr.CallWindow]'s within the projection list to
+ * [PartiqlAst.Expr.Id]'s that reference the new [PartiqlLogical.VarDecl].
+ */
+private class CallWindowFunctionProjectionReplacer() : VisitorTransformBase() {
+    val callWindowFunctionVisitorTransform = CallWindowFunctionReplacer()
+
+    override fun transformProjectItemProjectExpr_expr(node: PartiqlAst.ProjectItem.ProjectExpr): PartiqlAst.Expr {
+        return callWindowFunctionVisitorTransform.transformExpr(node.expr)
+    }
+
+    override fun transformProjectionProjectValue_value(node: PartiqlAst.Projection.ProjectValue): PartiqlAst.Expr {
+        return callWindowFunctionVisitorTransform.transformExpr(node.value)
+    }
```

Comment:
As mentioned in other comment, this is most likely where we need to make sure we don't recurse into exprSelect.